superset-frontend/plugins/plugin-chart-handlebars/src/plugin/controls/handlebarTemplate.tsx

```diff
@@ -0,0 +1,60 @@
+import {
+  ControlConfig,
+  ControlSetItem,
+  CustomControlConfig,
+  sharedControls,
+} from '@superset-ui/chart-controls';
+import { t, validateNonEmpty } from '@superset-ui/core';
+import React from 'react';
+import { CodeEditor } from '../../components/CodeEditor/CodeEditor';
+import { ControlHeader } from '../../components/ControlHeader/controlHeader';
+
+interface HandlebarsCustomControlProps {
+  value: string;
+}
+
+const HandlebarsTemplateControl = (
+  props: CustomControlConfig<HandlebarsCustomControlProps>,
+) => {
+  const val = String(
+    props?.value ? props?.value : props?.default ? props?.default : '',
+  );
+
+  const updateConfig = (source: string) => {
+    props.onChange(source);
+  };
+  return (
+    <div>
+      <ControlHeader>{props.label}</ControlHeader>
+      <CodeEditor
+        theme="dark"
+        value={val}
+        onChange={source => {
+          updateConfig(source || '');
+        }}
+      />
+    </div>
+  );
+};
+const handlebarsTemplateControlConfig: ControlConfig<any> = {
+  ...sharedControls.entity,
+  type: HandlebarsTemplateControl,
+  label: t('Handlebars Template'),
+  description: t('A handlebars template that is applied to the data'),
+  default: `<ul class="data_list">
+    {{#each data}}
+      <li>{{this}}</li>
+    {{/each}}
+  </ul>`,
+  isInt: false,
+
+  validators: [validateNonEmpty],
+  mapStateToProps: ({ controls }) => ({
+    value: controls?.handlebars_template?.value,
```

Comment:
let's add
```typescript
{
  ...,
  renderTrigger: true,
}
```
here to avoid having to retrigger the data request every time the template is updated (this make the chart update in real time when updating the template 🎉 )